tests/integration/test_wpk/test_wpk_agent.py

```diff
@@ -0,0 +1,272 @@
+_config = ConfigObj(config_file_path)
+_agent_version = _config['VERSION']
+
+test_metadata = [
+    # 1. Upgrade from initial_version to v4.0.0
+    {
+        'protocol': PROTOCOL,
+        'initial_version': _agent_version,
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': DEFAULT_UPGRADE_SCRIPT,
+        'chunk_size': 16384,
+        'simulate_interruption': False,
+        'simulate_rollback': False,
+        'results': {
+            'upgrade_ok': True,
+            'result_code': 0,
+            'receive_notification': True,
+            'status': 'Done',   
+        }
+    },
+    # 2. False upgrade script parameter
+    {
+        'protocol': PROTOCOL,
+        'initial_version': _agent_version,
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': 'fake_upgrade.sh',
+        'chunk_size': 16384,
+        'simulate_interruption': False,
+        'simulate_rollback': False,
+        'results': {
+            'upgrade_ok': False,
+            'error_message': 'err Could not chmod',
+            'receive_notification': False,
+        }
+    },
+    # 3. Simulate an interruption
+    {
+        'protocol': PROTOCOL,
+        'initial_version': _agent_version,
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': DEFAULT_UPGRADE_SCRIPT,
+        'chunk_size': 16384,
+        'simulate_interruption': True,
+        'simulate_rollback': False,
+        'results': {
+            'upgrade_ok': False,
+            'error_message': 'Request confirmation never arrived',
+            'receive_notification': False,
+        }
+    }
+]
+
+if _agent_version == 'v3.13.1':
+    test_metadata += [{
+        # 4. Simulate a rollback (v3.13.1)
+        'protocol': PROTOCOL,
+        'initial_version': 'v3.13.1',
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': DEFAULT_UPGRADE_SCRIPT,
+        'chunk_size': 16384,
+        'simulate_interruption': False,
+        'simulate_rollback': True,
+        'results': {
+            'upgrade_ok': True,
+            'result_code': 0,
+            'receive_notification': False,
+        }
+    }]
+elif _agent_version == 'v4.0.0':
+    test_metadata += [{
+        # 5. Simulate a rollback (v4.0.0)
+        'protocol': PROTOCOL,
+        'initial_version': 'v4.0.0',
+        'agent_version': 'v4.0.0',
+        'use_http': False,
+        'upgrade_script': DEFAULT_UPGRADE_SCRIPT,
+        'chunk_size': 16384,
+        'simulate_interruption': False,
+        'simulate_rollback': True,
+        'results': {
+            'upgrade_ok': True,
+            'result_code': 0,
+            'receive_notification': True,
+            'status': 'Failed',   
+        }
+    }]
+
+params = [
+    {
+        'CRYPTO': CRYPTO,
+        'SERVER_ADDRESS': SERVER_ADDRESS,
+        'REMOTED_PORT': 1514,
+        'PROTOCOL': PROTOCOL
+    } for x in range(0, len(test_metadata))
```

Comment:
When in a for loop we do not use the variable, it is usually defined as '_'.
```suggestion
    } for _ in range(0, len(test_metadata))
```